zerver/actions/channel_folders.py

```diff
@@ -0,0 +1,43 @@
+from django.db import transaction
+from django.utils.timezone import now as timezone_now
+
+from zerver.lib.channel_folders import get_channel_folder_dict, render_channel_folder_description
+from zerver.models import ChannelFolder, RealmAuditLog, UserProfile
+from zerver.models.realm_audit_logs import AuditLogEventType
+from zerver.models.users import active_user_ids
+from zerver.tornado.django_api import send_event_on_commit
+
+
+@transaction.atomic(durable=True)
+def check_add_channel_folder(
+    name: str, description: str, *, acting_user: UserProfile
+) -> ChannelFolder:
+    realm = acting_user.realm
```

Comment:
`realm` would be better done as a first independent parameter. From a readability standpoint, having the first parameter be what realm to add the channel folder to is good, and how we write other code.

Reply:
This can be a follow-up PR.